numpy/random/mtrand/mtrand.pyx

```diff
@@ -1551,18 +1551,17 @@ cdef class RandomState:
         cdef double flow, fhigh, fscale
         cdef object temp
 
-        flow = PyFloat_AsDouble(low)
-        fhigh = PyFloat_AsDouble(high)
+        if np.array(low).shape == np.array(high).shape == ():
```

Comment:
Yup, it is the Python thing, guess what confused me was the double `==`, but it does what it is expected to:

```Python
In [184]: 0 == 0 == 0
Out[184]: True

In [185]: (0 == 0) == 0
Out[185]: False

In [186]: 0 == (0 == 0)
Out[186]: False
```